src/workerd/api/sockets.c++

```diff
@@ -169,7 +171,7 @@ jsg::Ref<Socket> Socket::startTls(jsg::Lock& js, jsg::Optional<TlsOptions> tlsOp
   //
   // Detach the AsyncIoStream from the Writable/Readable streams and make them unusable.
   writable->removeSink(js);
-  readable->detach(js);
+  readable = readable->detach(js, true);
```

Comment:
Nit: since we're going with the `bool` argument here, would you mind adding a simple `/* ... */` style comment immediately after the `true` indicating what that is setting?